### PR TITLE
test(ci): switch all tests to sql backend

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by kwannoel** [comment]
Hi, before deprecating etcd backend, could we still keep their tests in `main-cron` still? Some users may continue using it and we won't deprecate etcd, at least in 2.0 I suppose.

**Comment by yezizp2012** [comment]
updates:
There's one flaky test in `end-to-end source test` when running `e2e_test/source/basic/kafka.slt `. Meta will encounter `Failed to acquire connection from pool: Connection pool timed out` as follows:
```
2024-10-10T05:31:58.00352696Z  WARN risingwave_meta::controller::cluster: Failed to load expire worker info from db error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:31:58.003630889Z  WARN risingwave_meta::hummock: Vacuum metadata error error=Hummock error: failed to access meta store: Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:32:28.005092227Z  WARN risingwave_meta::rpc::metrics: fail to list worker node error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:32:28.005118661Z  WARN risingwave_meta::rpc::metrics: fail to count worker node error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:32:58.006758545Z  WARN risingwave_meta::controller::cluster: Failed to load expire worker info from db error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:33:28.008841941Z  WARN risingwave_meta::rpc::metrics: fail to count worker node error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:33:28.008934887Z  WARN risingwave_meta::rpc::metrics: fail to list worker node error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:33:58.010520665Z  WARN risingwave_meta::controller::cluster: Failed to load expire worker info from db error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:33:58.010616525Z  WARN risingwave_meta::hummock: Vacuum metadata error error=Hummock error: failed to access meta store: Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:34:28.011121148Z  WARN risingwave_meta::rpc::metrics: fail to count worker node error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:34:28.011150804Z  WARN risingwave_meta::rpc::metrics: fail to list worker node error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:34:58.012202046Z  WARN risingwave_meta::controller::cluster: Failed to load expire worker info from db error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:35:28.013878678Z  WARN risingwave_meta::rpc::metrics: fail to count worker node error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:35:28.013962782Z  WARN risingwave_meta::rpc::metrics: fail to list worker node error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:35:58.015341782Z  WARN risingwave_meta::controller::cluster: Failed to load expire worker info from db error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:35:58.015431101Z  WARN risingwave_meta::hummock: Vacuum metadata error error=Hummock error: failed to access meta store: Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:36:28.016727587Z  WARN risingwave_meta::rpc::metrics: fail to count worker node error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:36:28.016763508Z  WARN risingwave_meta::rpc::metrics: fail to list worker node error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:36:58.018054737Z  WARN risingwave_meta::controller::cluster: Failed to load expire worker info from db error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:37:28.019971563Z  WARN risingwave_meta::rpc::metrics: fail to list worker node error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:37:28.019991183Z  WARN risingwave_meta::rpc::metrics: fail to count worker node error=Failed to acquire connection from pool: Connection pool timed out
2024-10-10T05:37:58.021876732Z  WARN risingwave_meta::controller::cluster: Failed to load expire worker info from db error=Failed to acquire connection from pool: Connection pool timed out
```

I'm still investigation.

File: ci/scripts/run-e2e-test.sh [hunk 1]
```diff
@@ -27,7 +27,6 @@ if [[ $mode == "standalone" ]]; then
 fi
 
 if [[ $mode == "single-node" ]]; then
-  export RUST_MIN_STACK=4194304
```
Comment (fuyufjh) [inline ci/scripts/run-e2e-test.sh:30]: This is good to delete, but I'm curious why it's not needed any more?

Comment (yezizp2012) [reply ci/scripts/run-e2e-test.sh:30]: It's already exist in `common.sh`. 😃

**Pull request merged**